src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java

```diff
@@ -0,0 +1,520 @@
+        {
+            walkToBlock(item.getPosition());
+        }
+
+        incrementActionsDone();
+
+        return HERDER_DECIDE;
+    }
+
+    /**
+     * Find animals in area.
+     *
+     * @return the {@link List} of animals in the area.
+     */
+    public List<T> searchForAnimals()
+    {
+
+        if (this.getTargetableArea() != null)
+        {
+            return new ArrayList<>((world.getEntitiesWithinAABB(
+              getAnimalClass(),
+              this.getTargetableArea()
+            )));
+        }
+        return new ArrayList<>();
+    }
+
+    public int getMaxAnimalMultiplier()
+    {
+        return 2;
+    }
+
+    /**
+     * Find items in hut area.
+     *
+     * @return the {@link List} of {@link EntityItem} in the area.
+     */
+    public List<EntityItem> searchForItemsInArea()
+    {
+        if (this.getTargetableArea() != null)
+        {
+            return new ArrayList<>((world.getEntitiesWithinAABB(
+              EntityItem.class,
+              this.getTargetableArea()
+            )));
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Get the Animal's class from the none Abstract.
+     */
+    public abstract Class<T> getAnimalClass();
+
+    /**
+     * Creates a simple area around the Herder's Hut used for AABB calculations for finding animals.
+     *
+     * @return The {@link AxisAlignedBB} of the Hut Area
+     */
+    private AxisAlignedBB getTargetableArea()
+    {
+        if (getOwnBuilding() == null)
+        {
+            return null;
+        }
+
+        return getOwnBuilding().getTargetableArea(world);
+    }
+
+    /**
+     * Lets the herder walk to the animal.
+     *
+     * @return true if the herder is walking to the animal.
+     */
+    public boolean walkingToAnimal(final EntityAnimal animal)
+    {
+
+        if (animal != null)
+        {
+            worker.setLatestStatus(new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_HERDER_GOINGTOANIMAL));
+            return walkToBlock(animal.getPosition());
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    /**
+     * Breed two animals together!
+     *
+     * @param animalOne the first {@link EntityAnimal} to breed.
+     * @param animalTwo the second {@link EntityAnimal} to breed.
+     */
+    private void breedTwoAnimals(final EntityAnimal animalOne, final EntityAnimal animalTwo)
+    {
+
+        final List<EntityAnimal> animalsToBreed = new ArrayList<>();
+        animalsToBreed.add(animalOne);
+        animalsToBreed.add(animalTwo);
+
+        for (final EntityAnimal animal : animalsToBreed)
+        {
+            if (!animal.isInLove() && !walkingToAnimal(animal))
+            {
+                //noinspection ConstantConditions
+                animal.setInLove(null);
+                worker.swingArm(EnumHand.MAIN_HAND);
+                new InvWrapper(getInventory()).extractItem(getItemSlot(getBreedingItem()), 1, false);
```

Comment:
Use inventory utils when manipulating Inventories!

Reply:
okay, i created a method in inventory utils (there was only a method to remove a list of itemStacks but not a singular one)